File: app/src/main/java/statusbar/finder/DatabaseHelper.java
package statusbar.finder;

import android.content.Context;
import app.cash.sqldelight.db.SqlDriver;
import app.cash.sqldelight.driver.android.AndroidSqliteDriver;
import lombok.Getter;

public class DatabaseHelper {
    private static final String DATABASE_NAME = "lyric.db";
    @Getter
    private static LyricDatabase database;

    public static synchronized void init(Context context) {
        if (database != null) {return;}
        SqlDriver driver = new AndroidSqliteDriver(LyricDatabase.Companion.getSchema(), context, DATABASE_NAME);
        database = LyricDatabase.Companion.invoke(driver);
    }
}

File: app/src/main/java/statusbar/finder/DatabaseHelper.kt
package statusbar.finder

import android.content.Context
import app.cash.sqldelight.driver.android.AndroidSqliteDriver

/**
 * LyricGetterExt - statusbar
 * @description TODO: coming soon.
 * @author VictorModi
 * @email victormodi@outlook.com
 * @date 2025/2/9 13:42
 */
object DatabaseHelper {
    private const val DATABASE_NAME = "lyric.db"
    var database: LyricDatabase? = null
        private set

    @Synchronized
    fun init(context: Context?) {
        if (database != null || context == null) {
            return
        }
        val driver = AndroidSqliteDriver(LyricDatabase.Schema, context, DATABASE_NAME)
        database = LyricDatabase(driver)
    }
}
